be/src/vec/exec/volap_scan_node.cpp

```diff
@@ -2019,4 +1342,535 @@ Status VOlapScanNode::get_hints(TabletSharedPtr table, const TPaloScanRange& sca
+            if (value.data != nullptr) {
+                if constexpr (T == TYPE_CHAR || T == TYPE_VARCHAR || T == TYPE_STRING ||
+                              T == TYPE_HLL) {
+                    auto val = StringValue(value.data, value.size);
+                    if (is_fixed_range) {
+                        RETURN_IF_ERROR(change_value_range<true>(
+                                range, reinterpret_cast<void*>(&val),
+                                ColumnValueRange<T>::remove_fixed_value_range));
+                    } else {
+                        RETURN_IF_ERROR(change_value_range<true>(
+                                not_in_range, reinterpret_cast<void*>(&val),
+                                ColumnValueRange<T>::add_fixed_value_range));
+                    }
+                } else {
+                    if (is_fixed_range) {
+                        RETURN_IF_ERROR(change_value_range<true>(
+                                range, reinterpret_cast<void*>(const_cast<char*>(value.data)),
+                                ColumnValueRange<T>::remove_fixed_value_range));
+                    } else {
+                        RETURN_IF_ERROR(change_value_range<true>(
+                                not_in_range,
+                                reinterpret_cast<void*>(const_cast<char*>(value.data)),
+                                ColumnValueRange<T>::add_fixed_value_range));
+                    }
+                }
+                effect = true;
+            }
+        }
+    }
+
+    if (is_fixed_range ||
+        not_in_range.get_fixed_value_size() <= _max_pushdown_conditions_per_column) {
+        if (!is_fixed_range) {
+            // push down not in condition to storage engine
+            not_in_range.to_in_condition(_olap_filter, false);
+        }
+        *push_down = effect;
+    }
+    return Status::OK();
+}
+
+template <PrimitiveType T>
+Status VOlapScanNode::_normalize_is_null_predicate(VExpr* expr, VExprContext* expr_ctx,
+                                                   SlotDescriptor* slot, ColumnValueRange<T>& range,
+                                                   bool* push_down) {
+    RETURN_IF_PUSH_DOWN
+    auto is_null_checker = [&](const std::string& fn_name) { return fn_name == "is_null_pred"; };
+    auto is_not_null_checker = [&](const std::string& fn_name) {
+        return fn_name == "is_not_null_pred";
+    };
+    if (TExprNodeType::FUNCTION_CALL == expr->node_type()) {
+        if (_should_push_down_function(reinterpret_cast<VectorizedFnCall*>(expr),
+                                       is_null_checker)) {
+            auto temp_range = ColumnValueRange<T>::create_empty_column_value_range(
+                    slot->type().precision, slot->type().scale);
+            temp_range.set_contain_null(true);
+            range.intersection(temp_range);
+            *push_down = true;
+        } else if (_should_push_down_function(reinterpret_cast<VectorizedFnCall*>(expr),
```

Comment:
Seems we can just retain `temp_range.set_contain_null(true);` in the conditional body